test/com/amazon/ion/impl/lite/IonSequenceLiteSubListViewEqualityTests.java

```diff
@@ -0,0 +1,180 @@
+package com.amazon.ion.impl.lite;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import com.amazon.ion.IonSequence;
+import com.amazon.ion.IonSystem;
+import com.amazon.ion.IonValue;
+import com.amazon.ion.system.IonSystemBuilder;
+import junitparams.JUnitParamsRunner;
+import junitparams.Parameters;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotEquals;
+
+@RunWith(JUnitParamsRunner.class)
+public class IonSequenceLiteSubListViewEqualityTests {
+    static final IonSystem SYSTEM = IonSystemBuilder.standard().build();
+    private static List<Integer> INTS = Arrays.asList(0, 1, 2, 3, 4, 5, 6);
+    private static List<Integer> OTHER_INTS = Arrays.asList(0, 1, 2, 33, 4, 5, 6);
+    private static List<IonValue> ARRAY_LIST = newArrayList(INTS);
+    private static List<IonValue> OTHER_ARRAY_LIST = newArrayList(OTHER_INTS);
```

Comment:
Given that we have one type that deviates from the contract for equality, are there others? Do we need tests for Ion types that are not scalars, Symbols (sids and all that) and more importantly container types.

Reply:
I considered adding types that were not Ion ints here but in the end I felt that would be superfluous since it's the equality of `SubListView` being tested here, not the equality of the items inside it.